specification/common-types/resource-management/v5/networksecurityperimeter.json

```diff
@@ -0,0 +1,400 @@
+    "contact": {}
+  },
+  "paths": {},
+  "definitions": {
+    "PublicNetworkAccess": {
+      "type": "string",
+      "description": "Allow, disallow, or let network security perimeter configuration control public network access to the protected resource. Value is optional but if passed in, it must be 'Enabled', 'Disabled' or 'SecuredByPerimeter'.",
+      "enum": [
+        "Enabled",
+        "Disabled",
+        "SecuredByPerimeter"
+      ],
+      "x-ms-enum": {
+        "name": "PublicNetworkAccess",
+        "modelAsString": true,
+        "values": [
+          {
+            "value": "Enabled",
+            "description": "Allows public network access to the resource"
+          },
+          {
+            "value": "Disabled",
+            "description": "Disallows public network access to the resource"
+          },
+          {
+            "value": "SecuredByPerimeter",
+            "description": "The network security perimeter configuration rules allow or disallow public network access to the resource. Requires an associated network security perimeter."
+          }
+        ]
+      }
+    },
+    "NetworkSecurityPerimeterConfigurationListResult": {
+      "description": "Result of a list NSP (network security perimeter) configurations request.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "Array of network security perimeter results.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/NetworkSecurityPerimeterConfiguration"
+          }
+        },
+        "nextLink": {
+          "description": "The link used to get the next page of results.",
+          "type": "string",
+          "format": "uri"
+        }
+      }
+    },
+    "NetworkSecurityPerimeterConfiguration": {
+      "description": "Network security perimeter (NSP) configuration resource",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "./types.json#/definitions/ProxyResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/NetworkSecurityPerimeterConfigurationProperties"
+        }
+      }
+    },
+    "NetworkSecurityPerimeterConfigurationProvisioningState": {
+      "description": "Provisioning state of a network security perimeter configuration that is being created or updated.",
+      "enum": [
+        "Succeeded",
+        "Creating",
+        "Updating",
+        "Deleting",
+        "Accepted",
+        "Failed",
+        "Canceled"
+      ],
+      "type": "string",
+      "readOnly": true,
+      "x-ms-enum": {
+        "name": "NetworkSecurityPerimeterConfigurationProvisioningState",
+        "modelAsString": true
+      }
+    },
+    "NetworkSecurityPerimeterConfigurationProperties": {
+      "description": "Network security configuration properties.",
+      "type": "object",
+      "properties": {
+        "provisioningState": {
```

Comment:
It would be easiest for read-only properties if both the reference and the type used readOnly: true,  this is an allowed companion property to $ref in autorest: https://github.com/Azure/autorest/blob/main/docs/openapi/howto/%24ref-siblings.md

Reply:
just need to add the readOnly: true here.